torchqdynamics/sesolve/sesolve.py

```diff
@@ -0,0 +1,66 @@
+from typing import List, Literal, Optional, Tuple
+
+import torch
+import torch.nn as nn
+
+from ..odeint import odeint
+from ..solver import SolverOption
+from ..types import OperatorLike, TensorLike, TimeDependentOperatorLike, to_tensor
+
+
+def sesolve(
+    H: TimeDependentOperatorLike,
+    psi: OperatorLike,
+    t_save: TensorLike,
+    *,
+    exp_ops: Optional[List[OperatorLike]] = None,
+    save_states: bool = True,
+    gradient_alg: Literal[None, 'autograd', 'adjoint'] = None,
+    parameters: Optional[Tuple[nn.Parameter, ...]] = None,
+    solver: Optional[SolverOption] = None,
+) -> torch.Tensor:
+    # Args:
+    #     H: (b_H?, n, n)
+    #     psi: (b_psi?, n, 1)
+    #
+    # Returns:
+    #     (y_save, exp_save) with
+    #     - y_save: (b_H?, b_psi?, len(t_save), n, 1)
+    #     - exp_save: (b_H?, b_psi?, len(exp_ops), len(t_save))
+
+    # TODO: support density matrices too
+    # TODO: H is assumed to be time-independent from here (temporary)
+
```

Comment:
I think we should add a test on the shape of `psi0` at the beginning of the function. Users might be tempted to give initial kets of shape (n,) or (b,n). Both of these should raise an error.

Reply:
I added a TODO 👍🏼  I prefer to merge this quickly without waiting for a review of the small changes because we have quite a lot of stacked PRs waiting. Also I think we should discuss about arguments validation, namely what are the checks we want to implement on all arguments (the one you proposed is definitely necessary 👍🏼)